src/services/ViewService.js

```diff
@@ -187,15 +187,22 @@ function ViewService($q, $http, $rootScope, URLService, ElementService, UtilsSer
                 } catch (e) {
                 }
             }
-
-            $http.get(URLService.getViewsURL(reqOb))
-            .then(function(data) {
-                        results = data;
-                    }).finally(function() {
-                        CacheService.put(requestCacheKey, results);
-                        deferred.resolve(results);
-                        delete inProgress[key];
-                    });
+            $http.get(URLService.getViewElementIdsURL(reqOb))
+            .then(function(response) {
+                var data = response.data.elementIds;
+                toGet = toGet.concat(data);
+            }).finally(function() {
+                var toGetSet = new Set(toGet);
```

Comment:
can remove the outer http.get and just keep what's in this finally section

Reply:
Updated